### DISCUSSION TITLE
Add per-quest settings

### DISCUSSION THREAD

**Discussion by Nielkrokodil** [streetcomplete/StreetComplete#4445] [Ideas]
Some quests are quite spammy in my area with almost always the same answer, so i turned them off in the settings. For example surface on paths (almost always asphalt on sidewalks), surface on roads (also almost always asphalt). Or surveillance cameras on big train stations (is it still here quest). 

I would like to have the option to filter some tasks of specific quests out, but keep the quests enabled for more interesting ones, for example highway=unclassified or highway=path without footway=sidewalk. Also I would like to disable surveillance cameras on the "is it still here" quest.

**Comment by mnalis** [comment]
Original StreetComplete is geared towards novice users, so it is designed to be as simple and with as little configuration options as possible, so I guess it is _quite unlikely_ that such level of customization would be added here in the upstream StreetComplete. 

However, there are few viable options:

- [StreetComplete Expert Edition](https://github.com/Helium314/StreetComplete/) fork does implement such per-quest preferences for some quests (but not all), e.g.:

  ![small_Screenshot_20221003_145032_de westnordost streetcomplete expert](https://user-images.githubusercontent.com/156656/193580969-43edad73-77a6-40c9-9345-b95f16997ffb.jpg)

  While it is not implemented yet for your specific quests you mention here, it is more likely that [issue request there](https://github.com/Helium314/StreetComplete/issues) might be implemented there, if you decide to go that way. Customizing the quest does require that you know which tags exactly you're interested in (and how to combine them using `and`, `or`, `(`, `)`).

- Another alternative is of course always to fork your own version of StreetComplete and modify that to your fancy. It does not require big amount of programming knowledge for such smaller changes, but it is more work than the above option. Reading https://github.com/streetcomplete/StreetComplete/blob/master/CONTRIBUTING_A_NEW_QUEST.md is very helpful if you decide to go that way. It is easier than it might look -- I've done it before, without knowing any kotlin language for example (and using only GitHub web editor and actions for compiling, without the need to install anything). For example, you'd edit this `selection query` to only select what you want to show in quest:
https://github.com/streetcomplete/StreetComplete/blob/83596598b0feb44cbea07286d66cbc02dd9229cb/app/src/main/java/de/westnordost/streetcomplete/quests/surface/AddRoadSurface.kt#L13-L34

  Your variant might be **much** simpler, if you like. For example, changing the body of that query to just
`highway ~ unclassified|path and footway!=sidewalk and (!surface or surface older today -12 years)` would ask the quest **only** if:
  - `highway=*` tag has value of either `unclassified` _or_ `path`, and
  - `footway=*` does not have `sidewalk` value (`!` means `not`), and
  - `surface` tag is either missing, _or_ has been added more than 12 years ago (so it is time to resurvey)

  After editing the file, you'd just select `Actions` / `Build debug apk` on your github [fork](https://docs.github.com/en/get-started/quickstart/fork-a-repo), and after about 15 minutes download ready-made `.apk` to install on your phone, and check if it works as you wanted.

- as another workaround (which does not require using different version of StreetComplete, but does not _really_ accomplish your goals, however might help you cope with them), you can create two `quest presets` (in `Settings` / `Quest selection and display order` / three-vertical-dots-menu &#8285; / `Manage presets`) - **one** which has most quests enabled, and **another one** which has only quests you find very valuable in all situations. Then you can switch between them depending on how much time you have / whether you want to be bothered by  "spammy" quests at that specific time or area.

**Comment by Nielkrokodil** [comment]
Thank you @mnalis very much for your great answer. I will try the expert version.
